src/peer_manager.rs

```diff
@@ -250,20 +261,34 @@ impl PeerMap {
     }
 }
 
+/// (whether requires resource proof, own group)
+type ExpectJoinResult = Result<(bool, (Prefix<XorName>, Vec<PublicId>)), RoutingTableError>;
```

Comment:
I'm hoping we can reduce this type to `Result<(Prefix<XorName>, Vec<PublicId>), Error>` (see [comment below](https://github.com/maidsafe/routing/pull/1261#discussion_r92999421)) which Clippy may accept without any need to typedef it.